trie/trie_test.go

```diff
@@ -646,19 +645,27 @@ func makeAccounts(size int) (addresses [][20]byte, accounts [][]byte) {
 	// Create a realistic account trie to hash
 	addresses = make([][20]byte, size)
 	for i := 0; i < len(addresses); i++ {
-		for j := 0; j < len(addresses[i]); j++ {
-			addresses[i][j] = byte(random.Intn(256))
-		}
+		data := make([]byte, 20)
+		random.Read(data)
+		copy(addresses[i][:], data)
 	}
 	accounts = make([][]byte, len(addresses))
 	for i := 0; i < len(accounts); i++ {
 		var (
-			nonce   = uint64(random.Int63())
-			balance = new(big.Int).Rand(random, new(big.Int).Exp(common.Big2, common.Big256, nil))
-			root    = emptyRoot
-			code    = crypto.Keccak256(nil)
+			nonce = uint64(random.Int63())
+			root  = emptyRoot
+			code  = crypto.Keccak256(nil)
 		)
-		accounts[i], _ = rlp.EncodeToBytes(&account{nonce, balance, root, code})
+		// The big.Rand function is not deterministic with regards to 64 vs 32 bit systems,
+		// and will consume different amount of data from the rand source.
+		//balance = new(big.Int).Rand(random, new(big.Int).Exp(common.Big2, common.Big256, nil))
+		// Therefore, we instead just read via byte buffer
+		numBytes := random.Uint32() % 33 // [0, 32] bytes
+		balanceBytes := make([]byte, numBytes)
+		random.Read(balanceBytes)
```

Comment:
Randomizing the number of digits and then randomizing the digits makes for a very uneven distribution. You could just always read and set 32 bytes as the big Int will be normalized to remove leading zeros. I do understand that this is just a test and that it really doesn't matter, but just wanted to point it out :-). Also, another alternative to big.Int.Rand() is crypto/rand.Int() that works on bytes, not words, so it would be deterministic in this regard.

Reply:
> You could just always read and set 32 bytes as the big Int will be normalized to remove leading zeros.

Actually, that was my first replacement. However, the intent is to have rlp blobs of varying size, and thus have a lot of small (1-byte) values -- which basically would never ever happpen when using a random 32-byte array. So this is worse from a pure distribution perspective, but better in this particular instance.